### ISSUE TITLE
Optimise grid wrapper logic

### ISSUE THREAD

**Issue by dr3** [bbc/simorgh#1381]
**Is your feature request related to a problem? Please describe.**
We need to have a custom grid wrapper per article block to ensure they meet the UX spec, however currently, as of https://github.com/bbc/simorgh/pull/1376 we apply the wrapper to every single block, even if those surrounding it are also have the same grid constraint.

It would be good to 'normalise' the wrappers to reduce the number of DOM nodes and help improve page performance.

**Describe the solution you'd like**
The current implementation produces the following

```
<GridItemConstrainedMedium> 
    <p>some text</p>
</GridItemConstrainedMedium> 
<GridItemConstrainedMedium> 
    <p>some more text</p>
</GridItemConstrainedMedium> 
<GridItemConstrainedMedium> 
    <p>some other text</p>
</GridItemConstrainedMedium> 
```

Where possible, reduce the number of wrappers 

```
<GridItemConstrainedMedium> 
    <p>some text</p>
    <p>some more text</p>
    <p>some other text</p>
</GridItemConstrainedMedium> 
```

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/bbc/simorgh/labels/Refinement%20Needed)

**Comment by dr3** [comment]
Blocked on https://github.com/bbc/simorgh/issues/2563

**Comment by jamesbhobbs** [comment]
Closing in favour of this issue https://github.com/bbc/simorgh/issues/2313 (not implying it won't be done, just that one issue is clearer to track)
